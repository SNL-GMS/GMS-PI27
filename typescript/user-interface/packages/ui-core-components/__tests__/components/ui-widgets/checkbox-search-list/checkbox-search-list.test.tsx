import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';

import { CheckboxSearchList } from '../../../../src/ts/components/ui-widgets/checkbox-search-list/checkbox-search-list';
import type * as CheckboxSearchListTypes from '../../../../src/ts/components/ui-widgets/checkbox-search-list/types';

const checkboxItem1 = {
  id: '1',
  name: 'first',
  checked: true
};

const checkboxItem2 = {
  id: '2',
  name: 'second',
  checked: false
};

const checkboxItem3 = {
  id: '3',
  name: 'third',
  checked: false
};

const checkboxItems = [checkboxItem1, checkboxItem2, checkboxItem3];
const onCheckboxChecked = jest.fn();
const props: CheckboxSearchListTypes.CheckboxListProps = {
  items: checkboxItems,
  maxHeightPx: 300,
  onCheckboxChecked
};

describe('CheckboxSearchList', () => {
  it('to be defined', () => {
    expect(CheckboxSearchList).toBeDefined();
  });

  it('CheckboxSearchList renders', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const result = render(<CheckboxSearchList {...props} />);
    expect(result.container).toMatchSnapshot();
  });

  it('CheckboxSearchList functions and clicks', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const result = render(<CheckboxSearchList {...props} />);
    fireEvent.click(result.getAllByRole('checkbox')[0]);
    fireEvent.change(result.getByRole('searchbox'), { target: { value: '23' } });
    fireEvent.blur(result.getByRole('searchbox'));
    expect(onCheckboxChecked).toHaveBeenCalledWith('1', false);
  });
});
