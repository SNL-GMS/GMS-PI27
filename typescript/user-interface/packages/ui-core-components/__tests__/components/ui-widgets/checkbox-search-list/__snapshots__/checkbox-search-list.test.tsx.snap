// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxSearchList CheckboxSearchList renders 1`] = `
<div>
  <div
    class="checkbox-search-list"
    style="height: 300px;"
  >
    <input
      class="checkbox-search-list__search"
      placeholder="Search input"
      tabindex="0"
      type="search"
      value=""
    />
    <div
      class="checkbox-search-list-item"
    >
      <label
        class="bp5-control bp5-checkbox"
      >
        <input
          checked=""
          data-cy-station-checkbox="first"
          type="checkbox"
        />
        <span
          class="bp5-control-indicator"
        />
      </label>
      <span
        class="checkbox-search-list-item__label"
      >
        first
      </span>
    </div>
    <div
      class="checkbox-search-list-item"
    >
      <label
        class="bp5-control bp5-checkbox"
      >
        <input
          data-cy-station-checkbox="second"
          type="checkbox"
        />
        <span
          class="bp5-control-indicator"
        />
      </label>
      <span
        class="checkbox-search-list-item__label"
      >
        second
      </span>
    </div>
    <div
      class="checkbox-search-list-item"
    >
      <label
        class="bp5-control bp5-checkbox"
      >
        <input
          data-cy-station-checkbox="third"
          type="checkbox"
        />
        <span
          class="bp5-control-indicator"
        />
      </label>
      <span
        class="checkbox-search-list-item__label"
      >
        third
      </span>
    </div>
  </div>
</div>
`;
